CHANGELOG.md

```diff
@@ -1,5 +1,46 @@
 ## Latest
 
+## 0.6.4
+
+### Important
+- Following components will be deprecated and archived in next release:
+  - clickhouse-cli-client
+  - clickhouse-grpc-client
+- No more builds for non-lts Java versions - no more Java 9 release builds.
+- Lowest supported Java version will be 11.
+  - Java 11 support will be ended before the end of 2023. 
+  - It is recommended to use Java 21. 
+
+### New Features 
+- [Client-V1/Apache HTTP] More configuration parameters for connection management. Useful for tuning performance.
+(https://github.com/ClickHouse/clickhouse-java/pull/1771)
+    - com.clickhouse.client.config.ClickHouseClientOption#CONNECTION_TTL - to configure connection time-to-live
+    - com.clickhouse.client.http.config.ClickHouseHttpOption#KEEP_ALIVE_TIMEOUT - to configure keep-alive timeout
+    - com.clickhouse.client.http.config.ClickHouseHttpOption#CONNECTION_REUSE_STRATEGY - to configure how connection 
+are resued in the connection pool.
+    - Additionally switched to using LAX connection pool for Apache Connection Manager to improve performance 
+for concurrent requests.
+- [Client-V2] Connection pool configuration https://github.com/ClickHouse/clickhouse-java/pull/1766
+    - com.clickhouse.client.api.Client.Builder.setConnectionRequestTimeout - to configure connection request timeout.
+Important when there are no connections available in the pool to fail fast. 
+    - com.clickhouse.client.api.Client.Builder.setMaxConnections - total max connections per route.
+    - com.clickhouse.client.api.Client.Builder.setConnectionTTL - to limit connection live ignoring keep-alive from server.
+    - com.clickhouse.client.api.Client.Builder.setConnectionReuseStrategy - to configure how connections are used.
+Select FIFO to reuse connections evenly or LIFO (default) to reuse the most recently active connections.
+- [Client-V2] All operations are now executed in calling thread to avoid extra threads creation. 
+Async operations can be enabled by `com.clickhouse.client.api.Client.Builder.useAsyncRequests` (https://github.com/ClickHouse/clickhouse-java/pull/1767)
+- [Client-V2] Spring Demo Service as usage example (https://github.com/ClickHouse/clickhouse-java/pull/1765)
```

Comment:
Should we group the code and docs changes into relevant sections? it would help to skim through the file

Reply:
Moved to documentation section.